packages/api-client/src/hooks/useResponseBody.ts

```diff
@@ -1,37 +1,44 @@
 import { extractFilename } from '@/libs/extractAttachmentFilename'
-import { computed } from 'vue'
+import { computed, isRef, ref } from 'vue'
 import MimeType from 'whatwg-mimetype'
+import type { Ref } from 'vue'
 
 /**
  * Processes the response body of an HTTP request.
  * Extracts MIME type, attachment filename, and generates a data URL.
  */
 export function useResponseBody(props: {
-  data: unknown
-  headers: { name: string; value: string; required: boolean }[]
+  data: Ref<unknown> | unknown
+  headers:
+    | Ref<{ name: string; value: string; required: boolean }[]>
+    | { name: string; value: string; required: boolean }[]
 }) {
   const isBlob = (b: any): b is Blob => b instanceof Blob
 
+  // Handle both Ref and direct values
+  const dataRef = computed(() => (isRef(props.data) ? props.data.value : props.data))
+  const headersRef = computed(() => (isRef(props.headers) ? props.headers.value : props.headers))
```

Comment:
```suggestion
  const dataRef = toRef(props.data)
  const headersRef = toRef(props.headers)
```
I believe this should work as well